### ISSUE TITLE
Interface intersecting generic gives strange error while assigning properties

### ISSUE THREAD

**Issue by Dudeim** [Microsoft/TypeScript#27477]
**TypeScript Version:**  3.1.1

**Search Terms:** intersection generic error message

**Code**

```ts
// A *self-contained* demonstration of the problem follows...
// Test this by running `tsc` on the command-line, rather than through another build tool such as Gulp, Webpack, etc.
interface IExample {
    item: boolean;
    loading?: boolean;
    value: string;
}

function Test<T>(){
    const test: IExample & T = {
        loading: true,
        value: 'something',
        item: false
    };
}

```

**Expected behavior:**
Code should not give any errors

**Actual behavior:**
Typescript errors on assigning the properties of variable `test`. 
Error message: `Type 'true' is not assignable to type '(false & T["loading"]) | (true & T["loading"])'.`
This is a change from TS 3.0.3 where the same code gives no error.

**Playground Link:** 
[Playground link](http://www.typescriptlang.org/play/#src=interface%20IExample%20%7B%0D%0A%20%20%20%20item%3A%20boolean%3B%0D%0A%20%20%20%20loading%3F%3A%20boolean%3B%0D%0A%20%20%20%20value%3A%20string%3B%0D%0A%7D%0D%0A%0D%0Afunction%20Test%3CT%3E()%7B%0D%0A%20%20%20%20const%20test%3A%20IExample%20%26%20T%20%3D%20%7B%0D%0A%20%20%20%20%20%20%20%20loading%3A%20true%2C%0D%0A%20%20%20%20%20%20%20%20value%3A%20'something'%2C%0D%0A%20%20%20%20%20%20%20%20item%3A%20false%0D%0A%20%20%20%20%7D%3B%0D%0A%7D%0D%0A)

**Related Issues:** 
This issue looks somewhat similar, but way more specific combined with React.
#27473

**Comment by jack-williams** [comment]
A value is assignable to an intersection type `A & B` if it is assignable to both `A` and `B`. So in your example, you are saying that the object:

```ts
{ loading: true, value: 'something', item: false }
```
is assignable to parameter `T` which could be _any_ type! This can't be safe because `T` could be conflicting: `{ loading: string }`, not an object: `number`, or even the type of non-existing values: `never`!

**Comment by Dudeim** [comment]
Hmmm ok, makes sense. Has this been changed in typescript 3.1? As this error suddenly popped up in the codebase when going from ts 3.0.3 -> 3.1.1?
